src/test/java/seedu/sellsavvy/logic/commands/ordercommands/AddOrderCommandTest.java

```diff
@@ -1,23 +1,137 @@
+import static seedu.sellsavvy.logic.commands.personcommands.PersonCommandTestUtil.showPersonAtIndex;
+import static seedu.sellsavvy.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.sellsavvy.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.sellsavvy.testutil.TypicalOrders.ABACUS;
+import static seedu.sellsavvy.testutil.TypicalOrders.BLOCKS;
+import static seedu.sellsavvy.testutil.TypicalPersons.getTypicalAddressBook;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.logic.Messages;
+import seedu.sellsavvy.logic.commands.exceptions.CommandException;
 import seedu.sellsavvy.model.Model;
 import seedu.sellsavvy.model.ModelManager;
+import seedu.sellsavvy.model.UserPrefs;
+import seedu.sellsavvy.model.person.Person;
 
 public class AddOrderCommandTest {
 
-    //TODO: Remove this
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs()).createCopy();
+    }
+
+    @Test
+    public void execute_validPersonIndexUnfilteredList_success() {
+        AddOrderCommand addFirstOrder = new AddOrderCommand(INDEX_FIRST_PERSON, ABACUS);
+
+        Model expectedModel = model.createCopy();
+        Person personToAddUnder = expectedModel.getFilteredPersonList().get(0);
+        String firstExpectedMessage = String.format(AddOrderCommand.MESSAGE_SUCCESS,
+                personToAddUnder.getName(), Messages.format(ABACUS));
+        personToAddUnder.getOrderList().add(ABACUS);
+
+        // First order
+        assertCommandSuccess(addFirstOrder, model, firstExpectedMessage, expectedModel);
+
+        // Second order
+        AddOrderCommand addSecondOrder = new AddOrderCommand(INDEX_FIRST_PERSON, BLOCKS);
+        String secondExpectedMessage = String.format(AddOrderCommand.MESSAGE_SUCCESS,
+                personToAddUnder.getName(), Messages.format(BLOCKS));
+        personToAddUnder.getOrderList().add(BLOCKS);
+
+        assertCommandSuccess(addSecondOrder, model, secondExpectedMessage, expectedModel);
+    }
+
     @Test
-    public void execute_newOrder_success() { // not a legit test
-        Model model = new ModelManager();
+    public void execute_duplicateOrder_warningGiven() {
+        AddOrderCommand addOrderCommand = new AddOrderCommand(INDEX_FIRST_PERSON, ABACUS);
+
         try {
-            assertNull(new AddOrderCommand().execute(model));
-        } catch (Exception e) {
+            addOrderCommand.execute(model);
+
+            Model expectedModel = model.createCopy();
+            Person personToAddUnder = expectedModel.getFilteredPersonList().get(0);
+            personToAddUnder.getOrderList().add(ABACUS);
+            String expectedMessage = String.format(MESSAGE_DUPLICATE_WARNING + AddOrderCommand.MESSAGE_SUCCESS,
```

Comment:
Should we rename MESSAGE_DUPLICATE_WARNING to MESSAGE_DUPLICATE_ORDER_WARNING?

Reply:
will rename